### PR TITLE
`base_url` usage on landing page

### PR DESCRIPTION
**Pull request by zachcoleman**
**Related Issue(s):**

- #

**Description:**

Adds the `base_url` usage for two links where it was missing. 

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by vincentsarago** [comment]
@zachcoleman thanks for your PR, I need to catch up with stac-fastapi but is there any tests for this or should we make one?

**Comment by zachcoleman** [comment]
> @zachcoleman thanks for your PR, I need to catch up with stac-fastapi but is there any tests for this or should we make one?

This functionality as it just matches the same functions other uses seems like it'd just be adding tests for sake of adding tests. It will be more useful to probably test `get_base_url`, as it's not tested today.

**Review by vincentsarago — APPROVED** [review]

**Comment by vincentsarago** [comment]
FYI: this ended up introducing a bug https://github.com/stac-utils/stac-fastapi-pgstac/pull/101/files#r1576403075

File: stac_fastapi/types/stac_fastapi/types/core.py [hunk 1]
```diff
@@ -581,9 +581,7 @@ async def landing_page(self, **kwargs) -> stac_types.LandingPage:
                 "rel": "service-desc",
                 "type": "application/vnd.oai.openapi+json;version=3.0",
                 "title": "OpenAPI service description",
-                "href": urljoin(
-                    str(request.base_url), request.app.openapi_url.lstrip("/")
-                ),
+                "href": urljoin(base_url, request.app.openapi_url.lstrip("/")),
```
Comment (vincentsarago) [inline stac_fastapi/types/stac_fastapi/types/core.py:584]: When a user adds a router_prefix to the API, the `base_url` as the prefix added in https://github.com/stac-utils/stac-fastapi/blob/f54227871c29cf15ae49b2ab96b28f357de60485/stac_fastapi/types/stac_fastapi/types/requests.py#L6-L14

The issue is that, when we create the fastapi app 

https://github.com/stac-utils/stac-fastapi/blob/f54227871c29cf15ae49b2ab96b28f357de60485/stac_fastapi/api/stac_fastapi/api/app.py#L75-L85

we set the openapi_url to either `/api` or to what the user has set the `openapi_url` settings.

so by default we register the openapi to `/api` but here when we do `urljoin(base_url, request.app.openapi_url.lstrip("/")),`, we endup with `/{base_url}/{router_prefix}/api`

**Pull request merged**